src/types.ts

```diff
@@ -39,3 +39,9 @@ export type Authentication =
   | InstallationTokenAuthentication
   | AppAuthentication
   | UserToServerAuthentication;
+
+export type Types = {
```

Comment:
This doesn't seem used :thinking:

Reply:
I don't think `Types` specifically is needed, but I kept it to avoid breaking the current API.

And the reason I moved it was to keep `index.ts` specifically dedicated to exporting code from this package. If you want tight control over what is exported from the package, then I think it's much clearer if the `index.ts` only has this:

```ts
export * from "./create-token-auth.js";
export type {
  Authentication,
  StrategyInterface,
  Token,
  Types,
} from "./types.js";
```

where each element is explicitly exported rather than this:

```ts
import { auth } from "./auth.js";
import { hook } from "./hook.js";
import type { StrategyInterface, Token, Authentication } from "./types.js";

export type Types = {
  StrategyOptions: Token;
  AuthOptions: never;
  Authentication: Authentication;
};

export const createTokenAuth: StrategyInterface = function createTokenAuth(
  token: Token,
) {
  if (!token) {
    throw new Error("[@octokit/auth-token] No token passed to createTokenAuth");
  }

  if (typeof token !== "string") {
    throw new Error(
      "[@octokit/auth-token] Token passed to createTokenAuth is not a string",
    );
  }

  token = token.replace(/^(token|bearer) +/i, "");

  return Object.assign(auth.bind(null, token), {
    hook: hook.bind(null, token),
  });
};
```

where there's a constant containing a function, some imports, a random exported type and you can't really instantly know everything that's exported at a glance.

But yeah I'd rather just delete that to be honest